sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java

```diff
@@ -163,91 +196,131 @@ public void addAssignedOverlayPort(long port) {
+                            .setExtract(false);
+                }
+            }
         }
 
         if (taskSpec.getDiscovery().isPresent()) {
             taskInfoBuilder.setDiscovery(getDiscoveryInfo(taskSpec.getDiscovery().get(), podInstance.getIndex()));
         }
 
+        if (useDefaultExecutor) {
+            Protos.ContainerInfo containerInfo = getContainerInfo(podInstance.getPod());
+            if (containerInfo != null) {
+                taskInfoBuilder.setContainer(containerInfo);
+            }
+        }
+
         setHealthCheck(taskInfoBuilder, serviceName, podInstance, taskSpec, taskSpec.getCommand().get());
         setReadinessCheck(taskInfoBuilder, serviceName, podInstance, taskSpec, taskSpec.getCommand().get());
 
         return taskInfoBuilder.build();
     }
 
-    private static Protos.ExecutorInfo.Builder getExecutorInfoBuilder(
+    private Protos.ExecutorInfo.Builder getExecutorInfoBuilder(
             String serviceName,
             PodInstance podInstance,
+            Protos.FrameworkID frameworkID,
             UUID targetConfigurationId,
             SchedulerFlags schedulerFlags) throws IllegalStateException {
         PodSpec podSpec = podInstance.getPod();
         Protos.ExecutorInfo.Builder executorInfoBuilder = Protos.ExecutorInfo.newBuilder()
                 .setName(podSpec.getType())
-                .setExecutorId(Protos.ExecutorID.newBuilder().setValue("").build()); // Set later by ExecutorRequirement
-        // Populate ContainerInfo with the appropriate information from PodSpec.
-        // This includes networks, rlimits, secret volumes...
-        Protos.ContainerInfo containerInfo = getContainerInfo(podSpec);
-        if (containerInfo != null) {
-            executorInfoBuilder.setContainer(containerInfo);
-        }
-
-        // command and user:
-        Protos.CommandInfo.Builder executorCommandBuilder = executorInfoBuilder.getCommandBuilder().setValue(
-                "export LD_LIBRARY_PATH=$MESOS_SANDBOX/libmesos-bundle/lib:$LD_LIBRARY_PATH && " +
-                        "export MESOS_NATIVE_JAVA_LIBRARY=$(ls $MESOS_SANDBOX/libmesos-bundle/lib/libmesos-*.so) && " +
-                        "export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jre*/) && " +
-                        // Remove Xms/Xmx if +UseCGroupMemoryLimitForHeap or equivalent detects cgroups memory limit
-                        "export JAVA_OPTS=\"-Xms128M -Xmx128M\" && " +
-                        "$MESOS_SANDBOX/executor/bin/executor");
+                .setExecutorId(Protos.ExecutorID.newBuilder().setValue("").build());
+        Protos.ContainerInfo.Builder containerBuilder = getContainerInfo(podSpec, true, false).toBuilder();
 
-        if (podSpec.getUser().isPresent()) {
-            executorCommandBuilder.setUser(podSpec.getUser().get());
-        }
+        executorInfoBuilder.getLabelsBuilder().addLabelsBuilder()
+                .setKey("DCOS_SPACE")
```

Comment:
I made a constant in `PodInfoBuilder` to handle this for now, deferring anything moving to `ExecutorLabelWriter` until after we're no longer bound to support both custom and default executors, since this entails some package visibility changes (don't want to have the scheduler package depend on the executor package, where those classes currently live, but moving them to common is outside the scope of this PR).